ccmlib/common.py

```diff
@@ -76,6 +77,36 @@ class UnavailableSocketError(CCMError):
     pass
 
 
+class LogPatternToVersion(object):
+
+    def __init__(self, versions_to_patterns, default_pattern):
+        self.versions_to_patterns = versions_to_patterns
+        if "0" not in self.versions_to_patterns:
```

Comment:
I see it. I just now realized we also need to compare `LooseVersion`s when reducing the keys for `keys_less_than_version`.